internal/v5/auth.go

```diff
@@ -294,6 +241,32 @@ func (h *ReqHandler) checkRequest(req *http.Request, entityIds []*router.Resolve
 	}, nil
 }
 
+// areAllowedEntities checks if all entityIds are in the allowedEntities list (space
+// separated).
+func areAllowedEntities(entityIds []*router.ResolvedURL, allowedEntities string) error {
+	allowedEntitiesMap := make(map[string]bool)
+	for _, curl := range strings.Fields(allowedEntities) {
+		allowedEntitiesMap[curl] = true
+	}
+	if len(entityIds) == 0 {
+		return errgo.Newf("API operation does not involve expected entity %v", allowedEntities)
+	}
+
+	for _, entityId := range entityIds {
+		if allowedEntitiesMap[entityId.URL.String()] {
+			continue
+		}
+		purl := entityId.PromulgatedURL()
+		if purl != nil {
+			if allowedEntitiesMap[purl.String()] {
+				continue
+			}
+		}
+		return errgo.Newf("API operation on entity %v not allowed", entityId.String())
```

Comment:
s/API //
?

(by definition we're talking to the API, so no need to include that test, I think)

Also, no need for the .String method call (it'll be called implicitly)